### PR TITLE
Make addon py2/py3 compatible

### PR DESCRIPTION
**Pull request by dumbo0001**
* Use future dependency to be able to write Python 3 code but be Python 2 compatible.
Make some changes to make addon work for Python 3 (Matrix) and Python 2 (older Kodi versions).

* Bump version to 2.3.0 in the icdrama addon and add script.module.future as dependency.

* Use Gujal repository in aznhusband repository to get latest of script.module.resolveurl.

* Remove python 2 dependency in the repository addon.xml (where zip files will be located).

### CODE REVIEW

**Comment by dumbo0001** [comment]
@soraxas I made some changes to make the addon py2/py3 compatible. 
You can have a check if you want.
Can you merge this in your branch so that it will be part of the pull request https://github.com/aznhusband/aznhusband.github.io/pull/55 that you've submitted to aznhusband?

**Review by boleanly — COMMENTED** [review]
I've tested and it works, but I'd suggest increasing the resolveurl min version to 5.1.0 to enforce Py 2/3 compatibility.

I've also looked at the changes in `actions.py`. It seems that BeautifulSoup is returning a Unicode string in Python 2 for the `iframe_src` variable. In Python 3, that just becomes a str object so no manipulation is actually needed there. In both Python 2/3, `base_url` is a str object, so you can replace lines 176 - 184 with just `iframe_url = str(iframe.attrs['src'])`. That will turn it into/keep as a str object in Python 2/3, allowing `urljoin` to be used.

File: src/plugin.video.icdrama/lib/actions.py [hunk 1]
```diff
@@ -169,7 +169,20 @@ def play_mirror(url):
...
+        base_url = config.base_url
+        iframe_src = iframe.attrs['src']
+        try:
+            base_url = config.base_url.decode('utf-8')
+        except (UnicodeDecodeError, AttributeError):
+            pass
+        try:
+            iframe_src = iframe.attrs['src'].decode('utf-8')
+        except (UnicodeDecodeError, AttributeError):
+            pass
```
Comment (boleanly) [inline src/plugin.video.icdrama/lib/actions.py]: You can replace lines 176 - 184 with just iframe_url = str(iframe.attrs['src']). That will turn it into/keep as a str object in Python 2/3, allowing urljoin to be used.

Haha I didn't know you can comment like this

Comment (dumbo0001) [reply src/plugin.video.icdrama/lib/actions.py]: 👍 Changes are made. I was too lazy to check where the problem was and Python is not my main programming language 😉

**Comment by dumbo0001** [comment]
> I've tested and it works, but I'd suggest increasing the resolveurl min version to 5.1.0 to enforce Py 2/3 compatibility.
> 
> I've also looked at the changes in `actions.py`. It seems that BeautifulSoup is returning a Unicode string in Python 2 for the `iframe_src` variable. In Python 3, that just becomes a str object so no manipulation is actually needed there. In both Python 2/3, `base_url` is a str object, so you can replace lines 176 - 184 with just `iframe_url = str(iframe.attrs['src'])`. That will turn it into/keep as a str object in Python 2/3, allowing `urljoin` to be used.

Min version of `script.module.resolveurl` has also been changed to `5.1.0`.

Fully tested successfully on clean install of Kodi 18 and 19 by installing from a test repository.
@soraxas can you merge it in your branch?

**Pull request merged**